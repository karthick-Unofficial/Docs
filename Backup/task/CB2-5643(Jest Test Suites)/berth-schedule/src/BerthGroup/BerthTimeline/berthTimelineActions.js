import * as t from "./actionTypes";
import { openEventForm } from "../../BerthToolbar/berthToolbarActions";
import { openManager } from "../../appActions";
import { selectManager } from "../../LookupManager/lookupManagerActions";
export {
	setNextDay,
	setPreviousDay
} from "../../DateControls/dateControlsActions";
export const loadFormData = data => {
	return {
		type: t.LOAD_FORM_DATA,
		payload: { data }
	};
};
export const editAssignment = ({ assignment }) => {
	return dispatch => {
		dispatch(openEventForm());
		dispatch(loadFormData(assignment));
	};
};

export const openBerthManager = () => {
	return dispatch => {
		dispatch(openManager());
		dispatch(selectManager("berths"));
	};
};
