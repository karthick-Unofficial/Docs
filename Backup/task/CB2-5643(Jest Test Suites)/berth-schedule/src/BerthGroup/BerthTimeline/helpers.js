import moment from "moment";

export const fromDay = (date, n) => {
	let newDate;
	if (n < 0) {
		newDate = moment(date).subtract(Math.abs(n), "d");
	} else {
		newDate = moment(date).add(n, "d");
	}
	return newDate.set({ hour: 0, minute: 0, second: 0 }).valueOf();
};

export const parseData = data => {
	const { schedule, vessel, footmark } = data;
	const { loa, id } = vessel;
	const { ata, atd, eta, etd } = schedule;
	const x0 = moment(ata || eta).valueOf();
	const x = moment(atd || etd).valueOf();

	const retval = [
		{
			key: id,
			x0,
			y0: footmark,
			x,
			y: footmark + loa,
			assignment: data
		}
	];
	return retval;
};

export const getShapeData = data => {
	const { schedule, vessel, footmark, footmarkAssignment } = data;
	const { loa } = vessel;
	const { ata, eta } = schedule;
	const top = footmark + loa - 12;
	const bottom = footmark + 12;
	const arrival = moment(ata || eta);
	const left = arrival.add(50, "m").valueOf();	// offset 50
	const center = arrival.add(50, "m").valueOf();	// offset 100
	const right = arrival.add(50, "m").valueOf();	// offset 150
	return footmarkAssignment === "stern"
		? [
			{ x: center, y: top },
			{ x: left, y: top - 60 },
			{ x: left, y: bottom },
			{ x: right, y: bottom },
			{ x: right, y: top - 60 }
		]
		: [
			{ x: left, y: top },
			{ x: left, y: bottom + 60 },
			{ x: center, y: bottom },
			{ x: right, y: bottom + 60 },
			{ x: right, y: top }
		];
};

export const getPrimaryLabels = data => {
	const labels = data.map(d => {
		const { schedule, vessel, footmark } = d;
		const { loa, name } = vessel;
		const { ata, eta } = schedule;
		return {
			x: moment(ata || eta)
				.add(360, "m")
				.valueOf(),
			yOffset: 0,
			y: footmark + loa / 2,
			label: name
		};
	});
	return labels;
};

export const getSecondaryLabels = data => {
	const labels = data.map(d => {
		const { schedule, vessel, footmark } = d;
		const { loa } = vessel;
		const { ata, eta } = schedule;
		return {
			x: moment(ata || eta)
				.add(360, "m")
				.valueOf(),
			yOffset: 16,
			y: footmark + loa / 2,
			label: `${loa}'`
		};
	});
	return labels;
};

export const getPendingPrimaryLabels = data => {
	const labels = data.map(d => {
		const { offset, schedule, vessel } = d;
		const { name } = vessel;
		const { ata, eta } = schedule;
		return {
			x: moment(ata || eta)
				.add(75, "m")
				.valueOf(),
			yOffset: 25,
			y: 60 + 50 * offset,
			label: name
		};
	});
	return labels;
};

export const getPendingSecondaryLabels = data => {
	const labels = data.map(d => {
		const { offset, schedule, vessel } = d;
		const { loa } = vessel;
		const { ata, eta } = schedule;
		return {
			x: moment(ata || eta)
				.add(75, "m")
				.valueOf(),
			yOffset: 43,
			y: 60 + 50 * offset,
			label: `${loa}'`
		};
	});
	return labels;
};

export const getTimeSpan = date => {
	return [fromDay(date, -1), fromDay(date, 6)];
};

export const getTimeGrid = date => {
	const scale = [-2, -1, 0, 1, 2, 3, 4, 5];
	const range = scale.map(n => fromDay(date, n));
	return range;
};
