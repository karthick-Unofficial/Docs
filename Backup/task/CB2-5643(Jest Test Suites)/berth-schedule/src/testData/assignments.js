const assignments = {
    "allAssignments": {},
    "dailyAssignments": {}
};

export { assignments };