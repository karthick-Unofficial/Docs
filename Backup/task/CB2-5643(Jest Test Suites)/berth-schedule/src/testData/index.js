export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { map } from "./map";
export { berths } from "./berths";
export { formPanel } from "./formPanel";
export { assignments } from "./assignments";
export { clientConfig } from "./clientConfig";