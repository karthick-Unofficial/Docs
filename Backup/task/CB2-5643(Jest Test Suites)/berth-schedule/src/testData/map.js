const map = {
    "open": false,
    "vessels": {},
    "subscriptions": []
};
export { map };