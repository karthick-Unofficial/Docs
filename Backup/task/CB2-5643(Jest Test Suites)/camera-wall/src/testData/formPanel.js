const formPanel = {
    "open": false,
    "editing": false,
    "data": {
        "agent": {
            "name": {
                "firstName": "",
                "lastName": ""
            },
            "email": "",
            "company": "",
            "phone": ""
        },
        "requestedBy": {
            "company": "",
            "name": {
                "firstName": "",
                "lastName": ""
            },
            "email": "",
            "phone": ""
        },
        "vessel": {
            "name": "",
            "mmsid": "",
            "imo": "",
            "loa": 0,
            "type": "",
            "grt": 0,
            "draft": 0
        },
        "barges": [],
        "cargo": [],
        "schedule": {
            "eta": null,
            "etd": null,
            "ata": null,
            "atd": null
        },
        "berth": {
            "id": "",
            "footmark": ""
        },
        "approved": false,
        "primaryActivity": {
            "id": ""
        }
    },
    "isGeneratingPdf": false,
    "exportingError": null
};

export { formPanel };