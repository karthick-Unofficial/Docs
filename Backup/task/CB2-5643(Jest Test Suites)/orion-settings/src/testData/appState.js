const appState = {
    "errors": {
        "newUserErrorMessage": null,
        "passwordChangeError": null,
        "shareDialogError": null,
        "createOrgError": null
    },
    "dialog": {
        "openDialog": null,
        "dialogData": null
    },
    "dock": {
        "dockData": {
            "WavCam": {}
        }
    },
    "sharingTokens": {
        "enabled": true
    },
    "viewing": {
        "selectedEntity": {
            "type": "org",
            "id": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b"
        }
    }
};
export { appState };