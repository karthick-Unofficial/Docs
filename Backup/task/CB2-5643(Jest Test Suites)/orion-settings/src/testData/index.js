export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { globalData } from "./globalData";