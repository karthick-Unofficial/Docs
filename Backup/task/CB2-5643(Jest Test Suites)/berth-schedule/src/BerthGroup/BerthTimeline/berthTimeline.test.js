import React from "react";
import BerthTimeline from "./BerthTimeline";
import { shallow } from "enzyme";
import * as reactRedux from 'react-redux';
import { assignments, berths, formPanel, i18n } from "../../testData";

jest.mock("react-redux", () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
}));

describe("BerthTimeline", () => {
    beforeEach(() => {
        useSelectorMock.mockImplementation(selector => selector(mockStore));
    })
    afterEach(() => {
        useSelectorMock.mockClear();
    })

    const useSelectorMock = reactRedux.useSelector;

    const mockStore = {
        assignments,
        date: "Fri Feb 03 2023 16:41:58 GMT+0530 (India Standard Time)",
        formPanel,
        berths,
        i18n
    };

    it("renders", () => {
        const wrapper = shallow(<BerthTimeline groupId="" />);
        expect(wrapper).toMatchSnapshot();
    });
});