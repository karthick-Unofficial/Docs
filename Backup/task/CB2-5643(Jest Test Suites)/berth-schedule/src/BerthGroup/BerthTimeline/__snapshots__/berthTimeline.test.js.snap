// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BerthTimeline renders 1`] = `ShallowWrapper {}`;
